odl/contrib/README.md

```diff
@@ -0,0 +1,95 @@
+
+
+## Procedure
+
+Since you do not have write access to the target location in the ODL repository, do the initial work on the code on [your personal fork](https://odl.readthedocs.io/dev/gitwash/forking_hell.html#forking). Within this fork, add your contribution to the `odl/contrib` directory. When the code is ready, the steps are as follows:
+
+- Submit a pull request. If you would like an in-depth review, please indicate so in writing or by choosing a reviewer on the Github pull request page.
+- We will check the pull request as a whole and (optionally) the code. Note that our Travis CI will not check anything in `odl.contrib`.
+- To keep the history clean and to avoid accidental bloating of the repository, we will squash your pull request into a single commit that summarizes the whole addition. You can choose to do that yourself and write [a good commit message](https://odl.readthedocs.io/dev/gitwash/development_workflow.html#the-commit-message).
+- If no in-depth review was asked for, we will simply merge the PR if everything is okay. Otherwise, merging is delayed until you indicate that you have modified the code according to review.
+
+That's all.
+
+
+## Package layout
+
+All code goes into a subdirectory, e.g. `mylib`, of the `odl/contrib` directory.
+If you have code examples, we strongly suggest that you put them into an `examples` directory in `odl/contrib/mylib/`. The reason for choosing this specific name is that any directory named `examples` is automatically ignored when running `pytest` (see below), which ensures that examples that open new windows, read from disk or take long time to run do not interfere with test runs.
+For unit tests, we recommend that you put them into `odl/contrib/mylib/test` or similar.
+
+In summary, the structure after adding your code would look like this:
+```
+ |
+ |---- odl
+ |      |
+ |      |-- contrib
+ |      |       |
+ |      ...     |-- __init__.py
+ |              |-- mylib
+ |              ...    |
+ |                     |-- __init__.py
+ |                     |-- mymodule1.py
+ |                     |-- mymodule2.py
+ |                     |-- examples
+ |                     |      |
+ |                     |      |-- example1.py
+ |                     |      |-- example2.py
+ |                     |      ...
+ |                     |
+ |                     |--- tests
+ |                     ...    |-- test_mymodule1.py
+ |                            |-- test_mymodule2.py
+ ...                          ...
+```
+Your package will then be importable as `from odl.contrib import mylib`.
+
+If your code extends existing ODL functionality, consider using the same directory structure. For instance, if you add a new geometry, you may want to put it into `odl/contrib/my_new_geometry/tomo/geometry/mygeometry.py` and add a modified ray transform to `odl/contrib/my_new_geometry/tomo/operators/ray_trafo.py`.
```

Comment:
Ah so that's what you meant? So replicate the tree directly in `odl/contrib` and then add custom stuff?
Not sure about that since it works against the isolated packages principle. I think it would make contributors feel less responsible for added stuff if it just becomes part of a bigger mass of stuff. If it lives in a directory on its own it's clear that it's their package, their responsibility, their credit.

I'm actually inclined to removing this recommendation altogether. I'd rather simply let them do whatever they want and just give comments when the code comes in.